cli/azd/pkg/tools/azcli/appconfig.go

```diff
@@ -0,0 +1,83 @@
+package azcli
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/appconfiguration/armappconfiguration"
+	"github.com/azure/azure-dev/cli/azd/pkg/convert"
+	"github.com/azure/azure-dev/cli/azd/pkg/identity"
+)
+
+type AzCliAppConfig struct {
+	Id         string `json:"id"`
+	Name       string `json:"name"`
+	Location   string `json:"location"`
+	Properties struct {
+		EnablePurgeProtection bool `json:"enablePurgeProtection"`
+	} `json:"properties"`
+}
+
+type AzCliAppConfigSecret struct {
+	Id    string `json:"id"`
+	Name  string `json:"name"`
+	Value string `json:"value"`
+}
+
+func (cli *azCli) GetAppConfig(ctx context.Context, subscriptionId string, resourceGroupName string, configName string) (*AzCliAppConfig, error) {
+	client, err := cli.createAppConfigClient(ctx, subscriptionId)
+	if err != nil {
+		return nil, err
+	}
+
+	config, err := client.Get(ctx, resourceGroupName, configName, nil)
+	if err != nil {
+		return nil, fmt.Errorf("getting app configuration: %w", err)
+	}
+
+	return &AzCliAppConfig{
+		Id:       *config.ID,
+		Name:     *config.Name,
+		Location: *config.Location,
+		Properties: struct {
+			EnablePurgeProtection bool "json:\"enablePurgeProtection\""
+		}{
+			EnablePurgeProtection: convert.ToValueWithDefault(config.Properties.EnablePurgeProtection, false),
+		},
```

Comment:
Can we update the model `AzCliAppConfig` and make the `Properties` type to be non-inline?

That way we don't need to duplicate the inline-declaration here.  We would just something like 
Properties: AzCliAppConfigProperties {
    EnablePurgeProtection: config.Properties.EnablePurgeProtection
}

Why do we need the `convert.ToValueWithDfault` ??  

According to the REST ref, `config.Properties.EnablePurgeProtection` is default to `false` (https://learn.microsoft.com/en-us/rest/api/appconfiguration/stable/configuration-stores/get?tabs=HTTP#definitions)
Are you getting `nil` instead of false?